sycl/include/sycl/detail/kernel_desc.hpp

```diff
@@ -277,6 +277,52 @@ template <typename KernelNameType> constexpr bool hasSpecialCaptures() {
   }
   return FoundSpecialCapture;
 }
+inline namespace compile_time_kernel_info_v1 {
+
+// This is being passed across ABI boundary, so we don't use std::string_view,
+// at least for as long as we support user apps built with GNU libstdc++'s
+// pre-C++11 ABI.
```

Comment:
After further considerations, I think we should do the right thing now and use `const sycl::detail::string_view` instead of `const char * const`. That would also mean that we have to introduce a new header for this class (since this header can't include any STL, and our `string_view` needs to interoperate with it).